src/v/cluster_link/model/types.h

```diff
@@ -118,23 +123,21 @@ struct connection_config
     using authn_variant = serde::variant<scram_credentials>;
     /// Authentication configuration for the connection
     std::optional<authn_variant> authn_config;
-    /// Path to certificate file to use
-    ss::sstring cert_file_path;
-    /// Path to key file (when mTLS is in use)
-    ss::sstring key_file_path;
-    /// Path to the CA file to use
-    ss::sstring ca_file_path;
+    /// certificate file to use
+    std::optional<tls_file_or_value> cert;
+    /// key to use (when mTLS is in use)
+    std::optional<tls_file_or_value> key;
+    /// The CA file to use
+    std::optional<tls_file_or_value> ca;
+    /// The client ID to use
+    ss::sstring client_id;
```

Comment:
This could be autogenerated by the cluster linking subsystem f.e. based on target cluster id. This way we will be able to identify the client logs related with a certain link